xclim/indices/_multivariate.py

```diff
@@ -733,7 +770,14 @@ def heat_wave_max_length(
     cond = (compare(tasmin, op, thresh_tasmin, constrain)) & (
         compare(tasmax, op, thresh_tasmax, constrain)
     )
-    max_l = cond.resample(time=freq).map(rl.longest_run, dim="time")
+    max_l = rl.resample_and_rl(
+        cond,
+        resample_before_rl,
+        rl.longest_run,
+        freq=freq,
+    )
+    # TODO: Should this step be integrated as an option in longest_run?
+
     out = max_l.where(max_l >= window, 0)
```

Comment:
```suggestion
    out = rl.resample_and_rl(
        cond,
        resample_before_rl,
        rl.rle_statistics,
        reducer='max',
        window=window,
        freq=freq,
    )
```
Not sure why the code was done this way. `rle.longest_run` is only a specific call to  `rle_statistics`which already exposes the needed functionality. Your comment made me notice it!